enterprise_catalog/apps/api/v1/views/enterprise_customer.py

```diff
@@ -119,8 +124,38 @@ def contains_content_items(self, request, enterprise_uuid, course_run_ids, progr
         }
         if (get_catalogs_containing_specified_content_ids or get_catalog_list):
             response_data['catalog_list'] = catalogs_that_contain_course
+
+        response_data[RESTRICTED_RUNS_ALLOWED_KEY] = self._get_restricted_runs_allowed_for_query(
+            course_run_ids, customer_catalogs,
+        )
         return Response(response_data)
 
+    def _get_restricted_runs_allowed_for_query(self, course_run_ids, customer_catalogs):
+        # filter the set of restricted course keys down to only
+        # those requested by the client, and only if those requested keys
+        # are top-level course keys (NOT course run keys).
+        requested_course_keys = {
+            key for key in course_run_ids if not key.startswith(COURSE_RUN_KEY_PREFIX)
+        }
+        serialized_data = defaultdict(lambda: defaultdict(dict))
+        for catalog in customer_catalogs:
+            if not catalog.restricted_runs_allowed:
+                continue
+            for restricted_course_key, restricted_runs in catalog.restricted_runs_allowed.items():
+                if restricted_course_key not in requested_course_keys:
+                    continue
+                course_dict = serialized_data[restricted_course_key]
+                for course_run_key in restricted_runs:
+                    run_dict = course_dict[course_run_key]
+                    if 'catalog_uuids' in run_dict:
+                        existing_uuids = set(run_dict['catalog_uuids'])
+                        existing_uuids.update(str(catalog.uuid))
+                        run_dict['catalog_uuids'] = list(existing_uuids)
+                    else:
+                        run_dict['catalog_uuids'] = [str(catalog.uuid)]
```

Comment:
Idea is that if the above change is adopted, you could just replace this whole section with:
```suggestion
                    run_dict['catalog_uuids'].add(str(catalog.uuid))
```

Reply:
oh yes, great idea!